src/views/CronTabForm/CronTabForm.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useState } from "react";
+import {
+  Form,
+  FormGroup,
+  TextInput,
+  ActionGroup,
+  Button,
+  PageSection,
+  Title,
+  Alert,
+  NumberInput,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+} from "@patternfly/react-core";
+import {
+  CRONTAB_APIGROUP_VERSION,
+  CRONTAB_APIVERSION,
+  CRONTAB_KIND,
+  CRONTAB_APIGROUP,
+} from "src/const";
+import {
+  useK8sModel,
+  k8sCreate,
+  useActiveNamespace,
+} from "@openshift-console/dynamic-plugin-sdk";
+import { useNavigate } from "react-router-dom-v5-compat";
+import { useCronTabTranslation } from "@crontab-utils/hooks/useCronTabTranslation";
+import { CronTabKind } from "@crontab-model";
+
+export const CronTabForm: React.FC = () => {
+  const [model] = useK8sModel({
+    version: CRONTAB_APIVERSION,
+    kind: CRONTAB_KIND,
+    group: CRONTAB_APIGROUP,
+  });
+  const [name, setName] = useState("");
+  const [cronSpec, setCronSpec] = useState("");
+  const [image, setImage] = useState("");
+  const [replicas, setReplicas] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+  const navigate = useNavigate();
+  const { t } = useCronTabTranslation();
+  const [namespace] = useActiveNamespace();
+
+  const onReplicasMinus = () => {
+    setReplicas((currentReplicas) => currentReplicas - 1);
+  };
+  const onReplicasPlus = () => {
+    setReplicas((currentReplicas) => currentReplicas + 1);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setLoading(true);
+    setError("");
+
+    const data: CronTabKind = {
+      apiVersion: CRONTAB_APIGROUP_VERSION,
+      kind: CRONTAB_KIND,
+      metadata: {
+        name,
+        namespace,
+      },
+      spec: {
+        cronSpec,
+        image,
+        replicas,
+      },
+    };
+
+    k8sCreate({ model, data })
+      .then(() => {
+        setLoading(false);
+        navigate(`/k8s/ns/${namespace}/stable.example.com~v1~CronTab`);
+      })
+      .catch((err) => {
+        setLoading(false);
+        setError(t("Error creating CronTab: {{err}}", { err: err.toString() }));
+      });
+  };
+  return (
+    <PageSection>
+      <Title headingLevel="h1">{t("Create Form")}</Title>
+      <Form>
+        <FormGroup label={t("Name")} fieldId="crontab-name" isRequired>
+          <TextInput
+            id="crontab-name"
+            name="name"
+            onChange={(_e, value) => setName(value)}
+            value={name || ""}
+            required
```

Comment:
It looks like only `Name` is required field for a CronTab resource? A CronTab resource can be successfully created with following YAML
```
apiVersion: stable.example.com/v1
kind: CronTab
metadata:
  name: my-new-cron-object
  namespace: default
```

Reply:
@yapei Could you open a bug for the issue on the details page, that is definitely NOT supposed to happen 😆

Reply:
@TheRealJon here is the bug https://issues.redhat.com/browse/OCPBUGS-56830 
we should either
1. mark all spec fields as required so that user can not create a CronTab successfully without spec 
2. add optional chaining so CronTab list page will always return correctly even spec is missing